pkg/ocp/machinesets.go

```diff
@@ -95,46 +95,51 @@ func (msd *k8sMachineSetDeployer) clientForMsd(nameSpace string) dynamic.Resourc
 	return msd.dynamicClient.Resource(machinesetGVR).Namespace(nameSpace)
 }
 
-func (msd *k8sMachineSetDeployer) GetWorkerNodeImage(workerNodeList []string, machineSet *unstructured.Unstructured,
+func (msd *k8sMachineSetDeployer) GetWorkerNodeImage(workerNodeNames []string, machineSet *unstructured.Unstructured,
 	infraID string,
 ) (string, error) {
 	machineSetClient, err := msd.clientFor(machineSet)
 	if err != nil {
 		return "", err
 	}
 
-	if len(workerNodeList) == 0 {
+	var workerNodes []unstructured.Unstructured
+
+	if len(workerNodeNames) == 0 {
 		nodeList, err := machineSetClient.List(context.TODO(), metav1.ListOptions{})
 		if err != nil {
 			return "", errors.Wrapf(err, "error listing the machineSets")
 		}
 
-		for _, machineName := range nodeList.Items {
-			workerNodeList = append(workerNodeList, machineName.GetName())
-		}
-	}
+		workerNodes = nodeList.Items
+	} else {
+		workerNodes = make([]unstructured.Unstructured, len(workerNodeNames))
 
-	for _, nodeName := range workerNodeList {
-		existing, err := machineSetClient.Get(context.TODO(), nodeName, metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			continue
-		}
+		for _, workerNodeName := range workerNodeNames {
+			workerNode, err := machineSetClient.Get(context.TODO(), workerNodeName, metav1.GetOptions{})
```

Comment:
This could be improved if field selectors supports “in” searches, but as far as I’m aware they only support exact matches.